src/test/java/com/google/sps/PersistentMatchRepositoryTest.java

```diff
@@ -0,0 +1,130 @@
+package com.google.sps;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.PersistentMatchRepository;
+import com.google.sps.data.User;
+import java.util.Collection;
+import java.util.HashSet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+
+/** */
+@RunWith(JUnit4.class)
+public final class PersistentMatchRepositoryTest {
+
+  private static final User USER_A = new User("User A");
+  private static final User MATCH_A = new User("Match A");
+  private static final User MATCH_B = new User("Match B");
+  private static final User MATCH_C = new User("Match C");
+
+  private LocalServiceTestHelper localHelper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+ @Mock private HttpServletRequest request;
+
+  @Mock private HttpServletResponse response; 
+
+  @Before
+  public void setUp() throws Exception {
+    USER_A.setId("123");
+    MATCH_A.setId("456");
+    MATCH_B.setId("789");
+    MATCH_C.setId("101111");
+    localHelper.setUp();
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    localHelper.tearDown();
+  }
+
+  @Test
```

Comment:
These tests provide great coverage for ensuring that calling addMatch() and removeMatch() will result in the correct number of matches returned using getMatchesForUser().

The next step is to verify that the matches that are returned from getMatchesForUser() have the right ids. It looks like this step is dependent on fetchUserWithId() being implemented. Maybe add a TODO to update these tests?

Reply:
Sounds good, added it to my list